src/with_ladj.jl

```diff
@@ -84,9 +84,19 @@ _with_ladj_on_mapped(map_or_bc::Function, y_with_ladj::Tuple{Any,Real}) = y_with
 function _with_ladj_on_mapped(map_or_bc::Function, y_with_ladj)
     y = map_or_bc(_get_y, y_with_ladj)
```

Comment:
I think this should be changed to
```suggestion
    y = map_or_bc(first, y_with_ladj)
```
The internal function is not needed and it is more idiomatic than the current `get_index`-based implementation.